### ISSUE TITLE
saving pdf breaks with invisible markers?

### ISSUE THREAD

**Issue by amueller** [matplotlib/matplotlib#8142]
```python
import matplotlib.pyplot as plt
import numpy as np
plt.plot(np.sin(np.linspace(-5, 5, 100)), 'v', c='none')
plt.savefig("asdf.pdf")
```
> TypeError: 'NoneType' object is not subscriptable

This happened to me because I was using matplotlib 1.5 code, where this was actually not invisible.
Changing  ``markeredgecolor='k'`` (which was the default before) fixes the problem. But that is a pretty surprising error and might hint at something going wrong somewhere?

In particular, png rendering works fine, only pdf errors.

**Comment by anntzer** [comment]
Fixed by #8850 (AFAICT, please request a reopen if not).

**Comment by amueller** [comment]
Fixed in 2.1.0 (or earlier).
